src/Core/Material/BlinnPhongMaterialModel.hpp

```diff
@@ -40,9 +40,56 @@ class RA_CORE_API BlinnPhongMaterialModel : public SimpleMaterialModel
     sample( Vector3 inDir, Vector3 normal, Vector3 tangent, Vector3 bitangent, Vector2 u ) override;
     Scalar PDF( Vector3 inDir, Vector3 outDir, Vector3 normal ) override;
 
-    /// DATA MEMBERS
+    inline Utils::Color getSpecularColor() const { return m_ks; }
+    inline Scalar getShininess() const { return m_ns; }
+    inline Scalar getDiffuseLuminance() const { return m_diffuseLuminance; }
+    inline Scalar getSpecularLuminance() const { return m_specularLuminance; }
+    inline std::string getTexSpecular() const { return m_texSpecular; }
+    inline std::string getTexShininess() const { return m_texShininess; }
+    inline std::string getTexNormal() const { return m_texNormal; }
+
+    inline void setDiffuseColor( Utils::Color color ) {
+        m_kd = color;
+        Vector3 rgbToLuminance { 0.2126_ra, 0.7152_ra, 0.0722_ra };
+        Scalar dIntensity   = m_kd.rgb().dot( rgbToLuminance );
+        Scalar sIntensity   = m_ks.rgb().dot( rgbToLuminance );
+        Scalar diffSpecNorm = std::max( 1_ra, dIntensity + sIntensity );
+
+        m_diffuseLuminance /= diffSpecNorm;
+        m_specularLuminance /= diffSpecNorm;
+        m_alpha = color.alpha();
+    }
+
+    inline void setSpecularColor( Utils::Color color ) {
+        m_ks = color;
+        Vector3 rgbToLuminance { 0.2126_ra, 0.7152_ra, 0.0722_ra };
+        Scalar dIntensity   = m_kd.rgb().dot( rgbToLuminance );
+        Scalar sIntensity   = m_ks.rgb().dot( rgbToLuminance );
+        Scalar diffSpecNorm = std::max( 1_ra, dIntensity + sIntensity );
+
+        m_diffuseLuminance /= diffSpecNorm;
+        m_specularLuminance /= diffSpecNorm;
```

Comment:
You should remove redundant code to help maintenance by putting these lines in a private method that update the sampling parameter and which is called from setDiffuseColor and from setSpecularColor.
I just realize that, right now, you do not use the textures, if present, for your computations. Just add a reminder in the documentation of your method so that we will add texture management later.